src/Facets/EverclearFacet.sol

```diff
@@ -0,0 +1,235 @@
+    struct EverclearData {
+        bytes32 receiverAddress;
+        uint256 nativeFee;
+        bytes32 outputAsset;
+        uint256 amountOutMin;
+        uint48 ttl;
+        bytes data;
+        uint256 fee;
+        uint256 deadline;
+        bytes sig;
+    }
+
+    /// Errors ///
+
+    /// @notice Reverts when the destination chain is not supported by Everclear
+    error UnsupportedEverclearChainId();
+
+    /// Constructor ///
+
+    /// @notice Constructor for the contract.
+    /// @param _feeAdapter Fee adapter address.
+    constructor(address _feeAdapter) {
+        // Validate fee adapter address to prevent misconfiguration that would break bridge functionality
+        if (address(_feeAdapter) == address(0)) {
+            revert InvalidConfig();
+        }
+        FEE_ADAPTER = IEverclearFeeAdapter(_feeAdapter);
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Everclear
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _everclearData Data specific to Everclear
+    function startBridgeTokensViaEverclear(
+        ILiFi.BridgeData memory _bridgeData,
+        EverclearData calldata _everclearData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        noNativeAsset(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _everclearData);
+    }
+
+    /// @notice Performs a swap before bridging via Everclear
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _everclearData Data specific to Everclear
+    function swapAndStartBridgeTokensViaEverclear(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        EverclearData calldata _everclearData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        noNativeAsset(_bridgeData)
+    {
+        uint256 actualAmountAfterSwap = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+
+        // If we got more than expected from the swap (positive slippage),
+        // send the extra back to the user. We cannot adjust amountOutMin
+        // because Everclear's signature validation includes the original value.
+        if (actualAmountAfterSwap > _bridgeData.minAmount) {
```

Comment:
There is question here that I need to check with BE team. Please lets keep this open until I resolve it.